vscode/src/chat/chat-view/ChatController.ts

```diff
@@ -934,27 +970,98 @@ export class ChatController implements vscode.Disposable, vscode.WebviewViewProv
+                            return `(repo:${repoName})`
+                        }
+                        case 'file': {
+                            const repoName =
+                                mention.remoteRepositoryName ||
+                                mention.repoName ||
+                                (await getFirstRepoNameContainingUri(mention.uri))
+
+                            const workspace = vscode.workspace.getWorkspaceFolder(mention.uri)
+                            if (!repoName || !workspace) {
+                                return null
+                            }
+
+                            const filePath = escapeRegExp(
+                                mention.uri.toString().split(`${workspace.name}/`)[1] || ''
+                            )
+
+                            if (!filePath || !repoName) {
+                                return null
+                            }
+
+                            return `(file:${filePath} repo:${repoName})`
+                        }
+                        case 'openctx':
+                            switch (mention.providerUri) {
+                                case REMOTE_FILE_PROVIDER_URI: {
+                                    const filePath = escapeRegExp(mention.mention?.data?.filepath || '')
+                                    const repoName = escapeRegExp(mention.mention?.data?.reponame || '')
+                                    if (!filePath || !repoName) {
+                                        return null
+                                    }
+                                    return `(file:${filePath} repo:${mention.mention?.data?.repoName})`
+                                }
+                                case REMOTE_DIRECTORY_PROVIDER_URI: {
+                                    const filePath = escapeRegExp(
+                                        mention.mention?.data?.directoryPath || ''
+                                    )
+                                    const repoName = escapeRegExp(mention.mention?.data?.repoName || '')
+                                    if (!filePath || !repoName) {
+                                        return null
+                                    }
+                                    //const query = `repo:${repoRe} file:${directoryRe}.*\/.* select:file.directory count:10`
+                                    return `(file:^${filePath} repo:${mention.mention?.data?.repoName})`
+                                }
+                                case REMOTE_REPOSITORY_PROVIDER_URI:
+                                    return `(repo:${mention.repoName})`
+                                default:
+                                    return null
+                            }
+                        default:
+                            return null
+                    }
+                })
+            )
+        ).filter(scope => !!scope) as string[]
```

Comment:
nit: we can make this type safe. (also, once we upgrade typescript this should no longer be necessary 🥳 )

```suggestion
        ).filter((scope): scope is string => !!scope);
```

Reply:
TIL